### ISSUE TITLE
Improve default Parse.User security

### ISSUE THREAD

**Issue by dblythy** [parse-community/parse-server#7292]
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Current Limitation

By default, when a new user is created, [it will have public R](https://github.com/parse-community/parse-server/blob/36de1db65dc3c2bbf8f8051ee1862a06ef41e7f9/src/RestWrite.js#L1377), unless overriden by a cloud function.

### Feature / Enhancement Description

Currently, to secure users, you need to create a cloud function. In my view, the reverse should be the case - that by default, the Parse.User ACL is R+W false, and if you want the "old" functionality, you should use a cloud function.

The changelog would be:

**Breaking Change:** Parse.User now defaults to public R false on signup. To achieve public read on sign up, use a cloud trigger:

```js
Parse.Cloud.beforeSave(Parse.User, ({object}) => {
  if (!object.existed()) {
    const acl = new Parse.ACL();
    object.setPublicReadAccess(true);
    object.setACL(acl);
  }
});
```

### Alternatives / Workarounds

Create a cloud function to prevent public R access on signup. It's my view this should be reversed - a cloud function shouldn't be needed to secure the user class.

**Comment by mtrezza** [comment]
Thanks for suggesting.

As we are moving towards phased depreciation, before this would become a breaking change, the following would need to be implemented:
- A Parse Server option that allows to define whether a new User should be public or private with the default being public (status quo)
- A "default deprecation warning" that the default will change in a future version (PR for this is in the works to standardize and centralize these deprecation warnings)
- A new Parse Server release needs to be published (planned for the coming week)
- Not earlier than 6 months after the new release has been published (12 months if the current release contains breaking changes that did not follow phased deprecation), the default can be changed and the deprecation warning removed

**Comment by dblythy** [comment]
Awesome, thanks for the detailed write up. This will be a good way to show the depreciation policy, considering it will only change a few lines of code

**Comment by dblythy** [comment]
Closed via #7319. On master, if you set `enforcePrivateUsers` to `true`, users ACL _won't_ have public read on signup.
